glide-core/src/client/value_conversion.rs

```diff
@@ -581,6 +582,31 @@ pub(crate) fn convert_to_expected_type(
                     )
                 }
             }
+
+            Value::Array(outer_array) if matches!(outer_array[0], Value::Array(_) | Value::Map(_) )  => {
+                // Helper function to check if an item contains a non-null `running_script`
+                fn contains_running_script(value: &Value) -> bool {
+                    match value {
+                        Value::Array(inner_array) => inner_array.get(1) != Some(&Value::Nil),
+                        Value::Map(inner_map) => inner_map.first().map(|(_, v)| v) != Some(&Value::Nil),
+                        _ => false,
+                    }
+                }
+
+                let mut converted_map: Option<Value> = None;
+
+                for inner_item in &outer_array {
+                    if contains_running_script(inner_item) {
+                        converted_map = Some(convert_to_expected_type(inner_item.clone(), Some(ExpectedReturnType::FunctionStatsReturnType))?);
+                        // we have found a response with an active running script
+                        // the server can only run a single script / function at a time
+                        break; // We can return this node's response
+                    }
+                }
+
+                // Return the converted map if found, otherwise convert the first item in the array
+                Ok(converted_map.unwrap_or_else(|| convert_to_expected_type(outer_array[0].clone(), Some(ExpectedReturnType::FunctionStatsReturnType)).unwrap()))
```

Comment:
Why panic here?  I think the `unwrap` can be avoided.

Reply:
You're right it seems wrong